scipy/stats/_stats_py.py

```diff
@@ -1121,10 +1112,7 @@ def moment(a, order=1, axis=0, nan_policy='propagate', *, center=None):
     xp = array_namespace(a)
     a, axis = _chk_asarray(a, axis, xp=xp)
 
-    if xp.isdtype(a.dtype, 'integral'):
-        a = xp.asarray(a, dtype=xp.float64)
-    else:
-        a = xp.asarray(a)
+    a = xp_broadcast_promote(a, force_floating=True, xp=xp)
```

Comment:
This PR downgrades the dtype from float64 to float32 when default_dtype is float32.

Reply:
Yup, but only for array API strict. And it's temporary until we discuss what we _want_ to have happen when `SCIPY_ARRAY_API=1` becomes the default and only option (https://github.com/scipy/scipy/pull/22695#discussion_r1997905891).